lib/topics.js

```diff
@@ -8,18 +8,37 @@ const SNS    = require('./sns');
 class Topic {
 
   constructor(server, topicName) {
-    this._server = server;
-    this.name    = topicName;
+    this._server     = server;
+    this.name        = topicName;
+    this.subscribers = [];
   }
 
 
   // Publish to a topic.
   publish(message) {
-    return this.clientPromise
-      .then(client => {
-        const stringMessage = JSON.stringify(message);
-        return client.publish(this.name, stringMessage);
+    const stringMessage = JSON.stringify(message);
+
+    if (process.env.NODE_ENV === 'test') {
+      // JSON encode/decode the message before passing to
+      // subscriber, same transformation as when using SNS.
+      const parsedMessage = JSON.parse(stringMessage);
+      setImmediate(() => {
+        this.subscribers.forEach(handler => handler(parsedMessage));
       });
+      return Promise.resolve();
+    } else {
+      return this.clientPromise
+        .then(client => client.publish(this.name, stringMessage));
+    }
```

Comment:
Yeah I think you'd use something designed for that use case like Redis pub/sub. I'm not sure it would even work with SNS unless Ironium begins to store a bunch of state (subscription tokens and IDs).

Should I still add unsubscribe as the counterpart of subscribe, the for-now testing facility?